little-books/src/containers/blog-description/blog-description.tsx

```diff
@@ -0,0 +1,120 @@
+        setEditBlogdata((prevState)=>{
+            return{
+                ...prevState,
+                editedBlogName:event.target.value
+            }
+        })
+    }
+
+    const onEditBlogDescriptionChange = (event)=>{
+
+        setEditBlogdata((prevState)=>{
+            return{
+                ...prevState,
+                editedBlogDescription:event.target.value
+            }
+        })
+    }
+
+    const scRef = useRef<HTMLElement>(null);
+
+    const imageGen = ()=>{
+        if(scRef.current){
+            const ssContainer:HTMLElement|null = scRef.current.querySelector('div:nth-child(1)');
+            if(ssContainer) 
+                toPng(ssContainer,{cacheBust:false})
+                .then((dataUrl)=>{
+                    const temp = document.createElement('a');
+                    temp.href = dataUrl;
+                    temp.download = "report-image.png";
+                    temp.click();
+                })}}
+
+    return(
+        <main ref={scRef} className={`${styles['blog-description-wrapper']} ${styles[themeValue]}`}>
+            {blogData ? 
+            <form onSubmit={onBlogEditSubmit}>
+                <div className="screen-shot-wrapper">
+                <div className={styles['img-holder']}>
+                    <img src={blogData.photo} alt='blog-img'/>
+                </div>
+                {
+                    editState ? 
+                    <div className={styles['form-section']}>
+                    <input onChange={onEditBlogTitleChange} value={editBlogData.editedBlogName}/>
+                    <textarea onChange={onEditBlogDescriptionChange} value={editBlogData.editedBlogDescription}/>
+                    </div>:
+                    <div>
+                        <h5 className={styles['blog-title']}>{blogData.title}</h5>
+                        <p className={styles['blog-details']}>{blogData.details}</p>
+                    </div>
+                }
+                </div>
+
+                <section className={styles['button-section']}>
+                    {
+                        editState ? 
+                        <>
+                        <Button buttonType='button' type='secondary-btn' size='small' onClickHandler={onEditBlogToggle}>{stringConst.blogDescriptionConst.buttons.CANCEL}</Button>
+                        <Button buttonType='submit' type='primary-btn' size='small'>{stringConst.blogDescriptionConst.buttons.SAVE}</Button>
+                        </>:
+                        <>
+                        <Button buttonType='button' type='secondary-btn' size='small' onClickHandler={onEditBlogToggle}>{stringConst.blogDescriptionConst.buttons.EDIT}</Button>
+                        <Button buttonType='button' type='primary-btn' size='small' onClickHandler={imageGen}>SC</Button>
+                        </>
+                    }
+
+                </section>
+            </form>
+            :
+            <p>Select a blog</p>}
```

Comment:
Strings can go into constants

Reply:
all the strings has been moved to constants